src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -429,7 +429,9 @@ private bool IsValidPatternDesignation(bool whenIsKeyword)
 
         private CSharpSyntaxNode ParseExpressionOrPatternForSwitchStatement()
         {
+            _termState |= TerminatorState.IsExpressionOrPatternInCaseLabelOfSwitchStatement;
             var pattern = ParsePattern(Precedence.Conditional, whenIsKeyword: true);
+            _termState ^= TerminatorState.IsExpressionOrPatternInCaseLabelOfSwitchStatement;
```

Comment:
Usually we use a `var saveTerm = _termState; _termState |= ...; _termState = saveTerm` pattern.

If we assume everybody setting a TerminatorState flag is also responsible for clearing it, then the XOR pattern is fine. But I would want to make sure Cyrus is fine with this.

Reply:
All logic of parsing patterns is scoped in this file. There are no writes to the `_termState` anywhere else in `LanguageParser_Patterns.cs`, so I did it this way. Considering that, it becomes more a style and consistency question than the acual logic one.

Reply:
er... this just gets confusing to me (i have to think about xor semantics).  In this case, would prefer to just keep the same as elsewhere.  Basically, i don't trust my own brain here :D